### PR TITLE
fix: account name checks FE-724

### PR DESCRIPTION
**Pull request by nelitow**
- Closes https://github.com/FuelLabs/fuels-wallet/issues/1263

Fixes:
- User not being able to change the account name to case-sensitive variations of the same word.
  - Example: Changing a case typo, like **Fuel wAllet** -> **Fuel Wallet** was not possible
- User not being able to change or create accounts that are substring of eachother.
  - Example: Changing **Fuel Wallet** -> **Wallet** was not possible

Considerations:
- The `filterByName` function sound like it could be used for a search filter, in which it should be case insensitive, but that is not the case at the moment, it is only used for account name editing and creation.
- Users can now create multiple accounts with similar names, which can cause confusion, changing only a letter case, so:
  - **Account 1** and **account 1** can now coexist as account names

UPDATE https://github.com/FuelLabs/fuels-wallet/pull/1570/commits/b611c8b9d1c08996bde6eb0ac7af21f9cdcbae30
Addressing an issue that arose as the function was used for account searching inside the wallet by another method, so now the account addition and editing don't rely on the same function. That was caught by e2e tests.

### CODE REVIEW

**Review by helciofranco — APPROVED** [review]
LGTM, your consideration about name conflicts is fine.

for reference, check rabby wallet, they allow even same exact name:
| Rabby Wallet |
| --- |
| Screenshot 2024-10-09 at 10 35 43 |

**Comment by helciofranco** [comment]
it's missing to add a changeset @nelitow 

`pnpm changeset`

File: packages/app/src/systems/Account/services/account.ts [hunk 1]
```diff
@@ -231,9 +231,9 @@ export class AccountService {
   }
 
   static filterByName(accounts: Account[], name = '') {
-    return accounts.filter((account) =>
-      account.name.toLowerCase().includes(name.toLowerCase())
-    );
+    return accounts.filter((account) => {
+      account.name === name;
```
Comment (helciofranco) [inline packages/app/src/systems/Account/services/account.ts:235]: it's missing a return clause here

```suggestion
      return account.name === name;
```

**Review by helciofranco — DISMISSED** [review]

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
this broke the test that validates connected screen flow.


the comparison before was done with `includes` and now it's done with `===` what was the reason for changing this? the issue was to fix the lowerCase only

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**